pymatgen/analysis/reaction_calculator.py

```diff
@@ -15,10 +15,10 @@
 from pymatgen.entries.computed_entries import ComputedEntry
 from monty.json import MontyDecoder
 from monty.fractions import gcd_float
+from uncertainties import ufloat
```

Comment:
I like this library, but is it necessary here (considering we would have to add as a core requirement)? It looks like the functionality of `uncertainties` is not used in this class beyond book-keeping, which could be done by a named_tuple or otherwise. I'd err towards keeping it but just wanted to clarify this point.

Reply:
Beyond book-keeping, we use `uncertainties` to calculate the propagated uncertainty when adding/subtracting uncertain quantities. It would be possible to code this ourselves (we had it that way in a previous iteration), but I feel its more robust to let the dedicated library handle it. Also, in some of our previous discussions I thought you were in favor of adding `uncertainties` as a core requirement, because it will facilitate some other functionality we have in the pipeline? In any event, I'd prefer to keep unless you strongly oppose. I think the package is pretty lightweight.

Reply:
Just to respond to this, yes I'm in favor of `uncertainties` via being in favor of `pint`, so I don't mind it getting added here, but mostly with a view of also addressing the unit question down the road. If this were the only place we'd use it I'd be more hesitant.